src/VirtoCommerce.CartModule.Data/Repositories/CartRepository.cs

```diff
@@ -66,8 +66,10 @@ public virtual async Task<ShoppingCartEntity[]> GetShoppingCartsByIdsAsync(strin
 
                         if (lineItemIds.Any())
                         {
+                            var lineItemTypeFullName = typeof(LineItem).FullName;
                             await TaxDetails.Where(x => lineItemIds.Contains(x.LineItemId)).LoadAsync();
                             await Discounts.Where(x => lineItemIds.Contains(x.LineItemId)).LoadAsync();
+                            await DynamicPropertyObjectValues.Where(x => x.ObjectType == lineItemTypeFullName && lineItemIds.Contains(x.LineItemId)).LoadAsync();

```

Comment:
Load dynamic properties need to check  if flag `WithDynamicProperties` is set

Reply:
Was done.